src/DockAreaTabBar.h.autosave

```diff
@@ -0,0 +1,251 @@
+	 */
+	int currentIndex() const;
+
+	/**
+	 * Returns the current tab or a nullptr if no tab is selected.
+	 */
+	CDockWidgetTab* currentTab() const;
+
+	/**
+	 * Returns the tab with the given index
+	 */
+	CDockWidgetTab* tab(int Index) const;
+
+	/**
+	 * Filters the tab widget events
+	 */
+	virtual bool eventFilter(QObject *watched, QEvent *event) override;
+
+	/**
+	 * This function returns true if the tab is open, that means if it is
+	 * visible to the user. If the function returns false, the tab is
+	 * closed
+	 */
+	bool isTabOpen(int Index) const;
+
+	/**
+	 * Overrides the minimumSizeHint() function of QScrollArea
+	 * The minimumSizeHint() is bigger than the sizeHint () for the scroll
+	 * area because even if the scrollbars are invisible, the required speace
+	 * is reserved in the minimumSizeHint(). This override simply returns
+	 * sizeHint();
+	 */
+	virtual QSize minimumSizeHint() const override;
+
+	/**
+	 * The function provides a sizeHint that matches the height of the
+	 * internal viewport.
+	 */
+	virtual QSize sizeHint() const override;
+
+public slots:
+	/**
+	 * This property sets the index of the tab bar's visible tab
+	 */
+	void setCurrentIndex(int Index);
+
+	/**
+	 * This function will close the tab given in Index param.
+	 * Closing a tab means, the tab will be hidden, it will not be removed
+	 */
+	void closeTab(int Index);
+
+signals:
+    /**
+     * This signal is emitted when the tab bar's current tab is about to be changed. The new
+     * current has the given index, or -1 if there isn't a new one.
+     */
+	void currentChanging(int Index);
+
+	/**
+	 * This signal is emitted when the tab bar's current tab changes. The new
+	 * current has the given index, or -1 if there isn't a new one
+	 */
+	void currentChanged(int Index);
+
+	/**
+	 * This signal is emitted when user clicks on a tab
+	 */
+	void tabBarClicked(int index);
+
+	/**
+	 * This signal is emitted when the close button on a tab is clicked.
+	 * The index is the index that should be closed.
+	 */
+	void tabCloseRequested(int index);
+
+	/**
+	 * This signal is emitted if a tab has been closed
+	 */
+	void tabClosed(int index);
+
+	/**
+	 * This signal is emitted if a tab has been opened.
+	 * A tab is opened if it has been made visible
+	 */
+	void tabOpened(int index);
+
+	/**
+	 * This signal is emitted when the tab has moved the tab at index position
+	 * from to index position to.
+	 */
+	void tabMoved(int from, int to);
+
+	/**
+	 * This signal is emitted, just before the tab with the given index is
+	 * removed
+	 */
+	void removingTab(int index);
+
+	/**
+	 * This signal is emitted if a tab has been inserted
+	 */
+	void tabInserted(int index);
+
+	/**
+	 * This signal is emitted when a tab title elide state has been changed
+	 */
+	void elidedChanged(bool elided);
+}; // class CDockAreaTabBar
+} // namespace ads
+//-----------------------------------------------------------------------------
+#endif // DockAreaTabBarH
+
```

Comment:
Remove this file :D

Reply:
yes, I just mentioned it in comparison page... do you know who creates it and how to prevent it in future?

Reply:
Qt Creator, the best way is to add `*.h.autosave` and `*.cpp.autosave` to the .gitignore file.

Reply:
I meant that you should not prevent this file from being created, it is how QtCreator manages backup saves.

Reply:
removed

Reply:
@rolivav Sorry, didn't see your answers earlier. Isn't `.gitignore` member of the repository, which, if I change it, will go to the next pull request for @githubuser0xFFFF? Isn't local `.git/info/exclude` better? 
Another question: why at all newly created files are added to the source control list without asking me? (with SVN Creator asked me to so and I could reject).